python/SWESimulators/CTCS.py

```diff
@@ -327,94 +326,115 @@ def __init__(self, cl_ctx, nx, ny, \
         self.ny_halo = np.int32(ny + 2*halo_y)
 
         # Load kernel for periodic boundary
-        self.boundaryKernels = Common.get_kernel(self.cl_ctx,\
-            "CTCS_boundary.opencl", block_width, block_height)
+        self.boundaryKernels = gpu_ctx.get_kernel("CTCS_boundary.cu", block_width, block_height)
+        
+        # Get CUDA functions and define data types for prepared_{async_}call()
+        self.boundaryUKernel_NS = self.boundaryKernels.get_function("boundaryUKernel_NS")
+        self.boundaryUKernel_NS.prepare("iiiiiiPi")
+        self.boundaryUKernel_EW = self.boundaryKernels.get_function("boundaryUKernel_EW")
+        self.boundaryUKernel_EW.prepare("iiiiiiPi")
+        self.boundaryVKernel_NS = self.boundaryKernels.get_function("boundaryVKernel_NS")
+        self.boundaryVKernel_NS.prepare("iiiiiiPi")
+        self.boundaryVKernel_EW = self.boundaryKernels.get_function("boundaryVKernel_EW")
+        self.boundaryVKernel_EW.prepare("iiiiiiPi")
+        self.boundaryEtaKernel_NS = self.boundaryKernels.get_function("boundaryEtaKernel_NS")
+        self.boundaryEtaKernel_NS.prepare("iiiiiiPi")
+        self.boundaryEtaKernel_EW = self.boundaryKernels.get_function("boundaryEtaKernel_EW")
+        self.boundaryEtaKernel_EW.prepare("iiiiiiPi")
+        self.boundary_linearInterpol_NS = self.boundaryKernels.get_function("boundary_linearInterpol_NS")
+        self.boundary_linearInterpol_NS.prepare("iiiiiiiiiiPi")
+        self.boundary_linearInterpol_EW = self.boundaryKernels.get_function("boundary_linearInterpol_EW")
+        self.boundary_linearInterpol_EW.prepare("iiiiiiiiiiPi")
+        self.boundary_flowRelaxationScheme_NS = self.boundaryKernels.get_function("boundary_flowRelaxationScheme_NS")
+        self.boundary_flowRelaxationScheme_NS.prepare("iiiiiiiiiiPi")
+        self.boundary_flowRelaxationScheme_EW = self.boundaryKernels.get_function("boundary_flowRelaxationScheme_EW")
+        self.boundary_flowRelaxationScheme_EW.prepare("iiiiiiiiiiPi")
 
         # Set kernel launch parameters
-        self.local_size = (block_width, block_height)
+        self.local_size = (block_width, block_height, 1)
         self.global_size = ( \
```

Comment:
This will launch a too large a grid, please double check all global sizes. 

In OpenCL global size should be the number of *threads* to launch, whilst 
in CUDA global size (grid size) should be the number of *blocks* to launch

Reply:
Oversight. Will fix.

Reply:
This is still not addressed.

Reply:
Fixed!